benchmark/benchmark_test.go

```diff
@@ -37,3 +37,21 @@ func BenchmarkPureFib(b *testing.B) {
 	}
 	result = r
 }
+
+func BenchmarkCollatz(b *testing.B) {
+	var r object.Object
+	program, env := testBuild(`let collatz = fn(n) { if (n==1) { 0 } else { if (n%2 == 0) { collatz(n/2) + 1 } else { collatz(3*n + 1) + 1 }}};let compute_sum_of_first_n_collatz = fn(n) { if (n== 1) { 0 } else { collatz(n) + compute_sum_of_first_n_collatz(n - 1) }}; compute_sum_of_first_n_collatz(100)`)
+	for i := 0; i < b.N; i++ {
+		r = evaluator.Eval(program, env)
+	}
+	result = r
```

Comment:
Did we ever end up validating if we need this variable assignment for accurate benchmarking?

Reply:
hmm yea we can toss this